### ISSUE TITLE
Host page example cannot be used programmatically without extra user action

### ISSUE THREAD

**Issue by guikubivan** [Microsoft/Office-Online-Test-Tools-and-Documentation#356]
We're trying to use the example hostpage from http://wopi.readthedocs.io/en/latest/hostpage.html, but since it suggests to use a form that is submitted programmatically, most browsers require the form submit to be triggered by an user event (a click, or in our case a mouseenter event).

Our application is displaying the form/iframe inside a pop-up dialog, and the form submit cannot be tied to the initial click to display the dialog. Therefore we end up needing an extra button(or mouseenter listener) to trigger the submission of the form. We tried to use a basic post request initially which would've worked better for us, but the request to urls such as https://word-view.officeapps-df.live.com gave us a CORS error.

It's worth noting our app is using Angular 2.1.1.

**Comment by chrisbro-MSFT** [comment]
That shouldn't be happening - browsers should not require user action for this form submit.  The sample code you cite is what we recommend to partners and is very similar to the code we use across multiple different Microsoft services.  And programmatic form posts are very common across the internet; they definitely work in general.

Browsers can change behavior significantly based on tiny differences in pages, however.  If you can create a reduced repro we'd like to look over it to help you.

**Comment by guikubivan** [comment]
You're right, it turned out we needed to wait for Angular to update the DOM before trying to submit the form (used a `setTimeout(fn, 0);`).

**Comment by chrisbro-MSFT** [comment]
I'm going to reopen this to use as a doc update, to mention this pattern & solution in the "Building a Host Page" documentation.
